src/main/java/fi/aalto/cs/apluscourses/model/APlusAuthentication.java

```diff
@@ -0,0 +1,17 @@
+package fi.aalto.cs.apluscourses.model;
+
+import org.jetbrains.annotations.NotNull;
+
+public class APlusAuthentication {
+  @NotNull
+  private String token;
+
+  public APlusAuthentication(@NotNull char[] token) {
+    this.token = String.copyValueOf(token);
```

Comment:
what would you think of storing the `token` itself as `char[]` as well ([some claim](https://www.baeldung.com/java-storing-passwords) it to be more secure) :lock:

Reply:
I thought about it, but we probably want to keep the token in memory as long as the IDE is running, so that the user doesn't have to enter it repeatedly. If the token is in memory as long as the IDE is running, then I'm not sure at what point we would actually overwrite the `char[]` token.

However, there's not much harm in storing it as `char[]`, so we can play it safe and do it anyways.